Makefile

```diff
@@ -21,3 +25,12 @@ verify-codegen-crds:
 verify-codegen: verify-codegen-crds
 verify: verify-codegen
 .PHONY: update-codegen-crds update-codegen verify-codegen-crds verify-codegen verify
+
+
+.PHONY: lint
+## Checks the code with golangci-lint
+lint: $(GOLANGCI_LINT_BIN)
+	$(GOLANGCI_LINT_BIN) run -c .golangci.yaml --deadline=30m
+
+$(GOLANGCI_LINT_BIN):
+	curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b ${BIN_DIR} ${GOLANGCI_LINT_VERSION}
```

Comment:
Installing from master seems dicy, since we could break as master becomes more strict.  Especially once we fork off of this branch into long-lived, hopefully quiet release branches, it would be nice to have this pinned to a specific version which we explicitly bump periodically.

Also, downloading a script and running it directly is fairly risky.  Can we use [hashpipe][1] or something similar to ensure we are only running the script we expect to get?

[1]: https://github.com/jbenet/hashpipe

Reply:
I've moved the installer script into the `hack` directory. Note that the installer script takes a version parameter for binary it downloads. But like you pointed out curl-piping a bad idea. But having the installation script in repo should be good enough. WDYT?

Reply:
that's still pulling both the checksum and the binary from the same location.  But either we can replace the script with something similar to pull our pinned version and compare with a local hash, or we can include a cache of the checksum file in our repo and twiddle the script to skip the checksum-fetching step so it uses our local copy.

Reply:
@wking I've changed the script so that the checksums are part of the script and the version is fixed. It only supports `linux` and `darwin` for now but should be enough for the time being.

Reply:
This looks good to me.

Reply:
Looks good to me to.

/hold cancel